backstory/src/main/java/com/google/sps/story/DatamuseRequestClient.java

```diff
@@ -0,0 +1,150 @@
+   */
+  public DatamuseRequestClient(String queryUrl) {
+    this.queryUrl = queryUrl;
+  }
+
+  /**
+   * Gets an array of adjectives related to the passed-in noun, which will have
+   * a passed-in cap on its size (array returned might be smaller if database
+   * does not have enough related adjectives). Fetches these adjectives
+   * by querying the Datamuse database.
+   *
+   * @param noun the noun to get adjectives related to (must be one word)
+   * @param cap the maximum number of adjectives to retrieve
+   * @return an array of adjectives related to the noun
+   * @throws IllegalArgumentException if noun is more than one word (has a space in it)
+   * @throws APINotAvailableException if Datamuse API cannot be reached
+   * @throws RuntimeException if JSON received back from the API cannot be parsed (JSONException)
+   *    or does not have objects of type JSON (ClassCastException)
+   */
+  public String[] getRelatedAdjectives(String noun, int cap)
+      throws IllegalArgumentException, APINotAvailableException, RuntimeException {
+    if (noun.contains(" ")) {
+      throw new IllegalArgumentException("Noun must be one word.");
+    }
+
+    // put together a query asking for adjectives related to the noun (rel_jjb=noun)
+    // and capping the number of results at 10 (max=10)
+    String query = queryUrl + "rel_jjb=" + noun + "&max=" + cap;
+    String jsonResponse = querySite(query);
+
+    // there are two try statements b/c there's a possibility for more than one JSONException to be
+    // thrown in this method so specifying what went wrong with that exact JSON exception
+    // will be more helpful hence a try block for each JSONException that needs a customized message
+    JSONArray jsonArray;
+
+    try {
+      jsonArray = new JSONArray(jsonResponse);
+    } catch (JSONException exception) {
+      throw new RuntimeException(
+          "Could not parse the JSON received back from the Datamuse Query.", exception);
+    }
+
+    try {
+      int length = jsonArray.length();
+      String[] adjectives = new String[length];
+
+      for (int i = 0; i < length; i++) {
+        JSONObject jsonObject = (JSONObject) jsonArray.get(i);
+        adjectives[i] = jsonObject.getString("word");
+      }
+
+      return adjectives;
+    } catch (ClassCastException exception) {
+      throw new RuntimeException("JSON array received was not of JSON objects.", exception);
+    } catch (JSONException exception) {
+      throw new RuntimeException(
+          "JSON Object did not have a String value for the key \"word\".", exception);
+    }
+  }
+
+  /**
+   * Query the site at the given url and retrieve the content (likely JSON) from that
+   * site. If there's nothing at that site, return null. This code was
+   * adapted from the DatamuseQuery class in the Datamuse4J repo
+   * (https://github.com/sjblair/Datamuse4J/blob/master/src/datamuse/DatamuseQuery.java).
+   *
+   * @param url the url to query
+   * @return the content at the site.
```

Comment:
Is this more specifically "the response body of the GET request"?

Reply:
It could be. As I mentioned above, I don't know if this is actually considered a GET request. I'm happy to make the change. Please let me know your thoughts after reading the last comment. @ykjchen

Reply:
Yes, this seems like it is. Thank you!

Reply:
Done